iml-manager-cli/src/filesystem.rs

```diff
@@ -7,22 +7,47 @@ use crate::{
         create_command, get, get_all, get_hosts, get_influx, get_one, wait_for_cmds_success,
         SendCmd, SendJob,
     },
-    display_utils::{generate_table, wrap_fut},
+    display_utils::{wrap_fut, DisplayType, IntoDisplayType as _},
     error::ImlManagerCliError,
     ostpool::{ostpool_cli, OstPoolCommand},
 };
+use console::Term;
 use futures::future::{try_join, try_join_all};
 use iml_wire_types::{Filesystem, FlatQuery, Mgt, Ost};
 use number_formatter::{format_bytes, format_number};
 use prettytable::{Row, Table};
 use std::collections::{BTreeMap, HashMap};
 use structopt::StructOpt;
 
+#[derive(Debug, serde::Serialize)]
+pub struct FilesystemAndStatsList(pub Vec<String>);
+
+impl IntoIterator for FilesystemAndStatsList {
+    type Item = String;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+#[derive(StructOpt, Debug)]
+pub struct ListItems {
+    /// Set the display type
+    ///
+    /// The display type can be one of the following:
+    /// tabular: display content in a table format
+    /// json: return data in json format
+    /// yaml: return data in yaml format
+    #[structopt(short = "d", long = "display", default_value = "tabular")]
+    display_type: DisplayType,
+}
+
 #[derive(Debug, StructOpt)]
 pub enum FilesystemCommand {
     /// List all configured filesystems
     #[structopt(name = "list")]
-    List,
+    List(ListItems),
```

Comment:
Don't we want to use it for other querying commands like "show", and not only for filesystems. In other words, make the display option global?

Reply:
The display has been updated for the following:
1. stratagem list
2. server list
3. ostpool list
4. filesystem list

Note that the way this is implemented varies between each of these and the default value should be tabular for each of these.

Reply:
>  In other words, make the display option global?

I don't think we can (reasonably) do it for all commands (yet). For example things like adding servers which interactively repaint the console would be tricky. I think it *would* make sense for show commands though.